### PR TITLE
Automation improvements.

### PR DESCRIPTION
**Pull request by LEGOL2**
- Fix CI jobs
- Add download button to readme
- Removed `clippy` and `fmt` requirement from pull requests

Resolves #
Does not resolve any issue. It's CI automated checks upgrade and user experience improvement.

## Proposed Changes
  - I've created workflows that actually work. Now each push and pull request will be tested and run with `clippy` and `fmt`
  - Not all users know what `git tag` is and how to find `download` section on Github. This massive, cool looking button is a link to release website, making it way easier for less advanced users.
  - `cargo fmt` and `cargo clippy` are no longer required in checklist below, as new CI will do it.

## Checklist

- [ ] Tested on all platforms changed
- [x] `cargo fmt` has been run on this branch
- [x] `cargo clippy` has been run on this branch
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
```
Comment (tarkah) [inline .github/workflows/continuous-integration.yml:18]: Can you please add `--all` to each of the cargo commands? We have some additional crates in the workspace that need to get checked as well

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
```
Comment (tarkah) [inline .github/workflows/continuous-integration.yml:32]: and here

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - run: rustup component add clippy
+      - uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: -- -D warnings
```
Comment (tarkah) [inline .github/workflows/continuous-integration.yml:64]: and here... `--all -- -D warnings`

**Comment by casperstorm** [comment]
Good additions, I however suggest we use the following button image:

download-button

Then it fits with the design on https://getajour.com - download the zipped image [here].(https://github.com/casperstorm/ajour/files/5313770/download-button.zip)

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (LEGOL2) [reply .github/workflows/continuous-integration.yml:32]: According to `cargo test --help`
```
--all                                        Alias for --workspace (deprecated)
--workspace                           Test all packages in the workspace
```
I will add `--workspace` flag.

Comment (tarkah) [reply .github/workflows/continuous-integration.yml:33]: Ahh yes, thanks!

**Pull request merged**